plugins/modules/aci_esg.py

```diff
@@ -0,0 +1,520 @@
+        ap=dict(type='str', aliases=['app_profile', 'app_profile_name']),
+        esg=dict(type='str', aliases=['name', 'esg_name']),
+        description=dict(type='str', aliases=['descr']),
+        epg=dict(type='str', aliases=['epg_name']),
+        admin_state=dict(type='str', default=False, choices=[False, True]),  # ESG Admin State
+        vrf=dict(type='str', aliases=['esg_vrf_name']),  # ESG VRF name
+        intra_esg_isolation=dict(
+            type='str',
+            default='unenforced',
+            choices=['enforced', 'unenforced'],
+        ),  # Intra ESG Isolation
+        preferred_group_member=dict(type='str', default='exclude', choices=['exclude', 'include']),  # Preferred Group Member
+        epg_selector_description=dict(type='str'),
+        match_key=dict(type='str'),  # ESG Tag Selector Key Name
+        value_operator=dict(
+            type='str', default='equals', choices=['contains', 'equals', 'regex']
+        ),  # ESG Tag Selector Operator type
+        match_value=dict(type='str'),  # ESG Tag Selector Key Value
+        tag_selector_description=dict(type='str'),
+        subnet_ip=dict(type='str', aliases=['esg_subnet_ip']),  # ESG Subnet IP
+        subnet_selector_description=dict(type='str'),
+        esg_contract_master=dict(type='str'),  # Contract Master name for ESG
+        state=dict(
+            type='str',
+            default='present',
+            choices=['absent', 'present', 'query'],
+        ),
+        name_alias=dict(type='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['tenant', 'ap', 'esg', 'vrf']],
+            ['state', 'present', ['tenant', 'ap', 'esg', 'vrf']],
+        ],
+        required_together=[['match_key', 'match_value']],
+    )
+
+    aci = ACIModule(module)
+    tenant = module.params.get('tenant')
+    ap = module.params.get('ap')
+    esg = module.params.get('esg')
+    description = module.params.get('description')
+    epg = module.params.get('epg')
+    admin_state = module.params.get('admin_state')
+    vrf = module.params.get('vrf')
+    intra_esg_isolation = module.params.get('intra_esg_isolation')
+    preferred_group_member = module.params.get('preferred_group_member')
+    state = module.params.get('state')
+    name_alias = module.params.get('name_alias')
+    epg_selector_description = module.params.get('epg_selector_description')
+    esg_contract_master = module.params.get('esg_contract_master')
+    value_operator = module.params.get('value_operator')
+    match_key = module.params.get('match_key')
+    match_value = module.params.get('match_value')
+    tag_selector_description = module.params.get('tag_selector_description')
+    subnet_selector_description = module.params.get('subnet_selector_description')
+    subnet_ip = module.params.get('subnet_ip')
+
+    if admin_state is True:
+        admin_state = 'yes'
+    else:
```

Comment:
What is the value of admin_state when it is not specified in the module attributes? Is it None or False? If it is None, I would prefer we don't set it to yes or no if it is not set by the user at all. Let the model default be applied by APIC.